frontend/src/slices/orders/order.slice.ts

```diff
@@ -0,0 +1,50 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+import { DataStatus } from '~/libs/enums/enums.js';
+import { type ValueOf } from '~/libs/types/types.js';
+import { type OrderResponseDto } from '~/packages/orders/orders.js';
+
+import { calculateOrderPrice, createOrder } from './actions.js';
+
+type State = {
+  orders: OrderResponseDto[];
+  price: number;
+  dataStatus: ValueOf<typeof DataStatus>;
+};
+
+const initialState: State = {
+  orders: [],
+  price: 0,
+  dataStatus: DataStatus.IDLE,
+};
+
+const { reducer, actions, name } = createSlice({
+  initialState,
+  name: 'orders',
+  reducers: {},
+  extraReducers(builder) {
+    builder
+      .addCase(createOrder.pending, (state) => {
```

Comment:
Wouldn't it be better to use **addMatcher** instead of **addCase** for all **pending** statuses at once? This should reduce the amount of code a bit.